src/platform/services-linux.c

```diff
@@ -34,13 +34,14 @@
  *
  * Linux implemention of services.
  */
-#ifdef __linux__
 
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
+#if defined(__linux__) || defined(__GLIBC__)
```

Comment:
The __GLIBC__ define comes from one of these headers, not from gcc, so the order does matter.  I'm not sure what happens on Linux libcs other than glibc, like musl or dietlibc.